tex/gregoriotex.lua

```diff
@@ -1088,121 +1085,90 @@ function lfs.dirname(oldpath)
-  end
-  local r,err = lfs.rmkdir(lfs.dirname(path))
-  if not r then
-    return nil,err.." (creating "..path..")"
-  end
-  return lfs.mkdir(path)
-end
--- end https://github.com/ToxicFrog/luautil/blob/master/lfs.lua
-
-local function clean_old_gtex_files(file_withdir)
-  local filename = ""
-  local dirpath = ""
-  local sep = ""
-  local onwindows = os.type == "windows" or
-    string.find(os.getenv("PATH"),";",1,true)
-  if onwindows then
-    sep = "\\"
-  else
-    sep = "/"
-  end
-  dirpath = string.match(file_withdir, "(.*)"..sep)
-  if dirpath then -- dirpath is nil if current directory
-    filename = "^"..file_withdir:match(".*/".."(.*)").."%-%d+_%d+_%d+[-%a%d]*%.gtex$"
-    if lfs.exists(dirpath) then
-      for a in lfs.dir(dirpath) do
+local function delete_versioned_files(dir, base, ext)
+  -- Assume that dir is either empty (current directory) or ends with separator
+  filename = "^"..base.."%-%d+_%d+_%d+[-%a%d]*%."..ext.."$"
+  if dir ~= "" then
+    if lfs.exists(dir) then
+      for a in lfs.dir(dir) do
         if a:match(filename) then
-          os.remove(dirpath..sep..a)
+          info("Deleting old file %s", dir..a)
+          os.remove(dir..a)
         end
       end
     end
   else
-    filename = "^"..file_withdir.."%-%d+_%d+_%d+[-%a%d]*%.gtex$"
     for a in lfs.dir(lfs.currentdir()) do
-      if a:match(filename) then os.remove(a) end
+      if a:match(filename) then
+        info("Deleting old file %s", a)
+        os.remove(a)
+      end
     end
   end
 end
 
+function table.extend(x, y)
+  table.move(y, 1, #y, #x+1, x)
+end
+
 local function compile_gabc(gabc_file, gtex_file, glog_file, allow_deprecated)
   info("compiling the score %s...", gabc_file)
-  local extra_args = ''
+  local cmd = {gregorio_exe()}
   if tex.count['gre@generate@pointandclick'] == 1 then
-    extra_args = extra_args..' -p'
+    table.insert(cmd, '-p')
   end
   if not allow_deprecated then
-    extra_args = extra_args..' -D'
+    table.insert(cmd, '-D')
   end
 
-  local cmd = string.format('%s %s -W -o %s -l %s "%s" 2> %s', gregorio_exe(),
-      extra_args, gtex_file, glog_file, gabc_file, glog_file)
-  res = os.execute(cmd)
+  table.extend(cmd, {'-W', '-o', gtex_file, '-l', glog_file, gabc_file})
+  info("running: %s", table.concat(cmd, ' '))
+  res = os.spawn(cmd)
+
   if res == nil then
     err("\nSomething went wrong when executing\n    '%s'.\n"
         .."shell-escape mode may not be activated. Try\n\n"
         .."%s --shell-escape %s.tex\n\n"
         .."See the documentation of Gregorio or your TeX\n"
         .."distribution to automatize it.",
-        cmd, tex.formatname, tex.jobname)
+        table.concat(cmd, ' '), tex.formatname, tex.jobname)
   elseif res ~= 0 then
-    local glog = io.open(glog_file, 'a+')
+    err("\nSomething went wrong when executing\n    '%s'",
+        table.concat(cmd, ' '))
+  else
+    -- Open glog_file for writing so that the LuaTeX recorder knows that gregorio wrote to it.
```

Comment:
Are we supposed to be opening glog?  The open statement opens gtex.

Reply:
Thank you, that's my fault